### ISSUE TITLE
Improve integration detection for Azure ASP.NET Core DataProtection packages

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#477]
During an Azure SDK NuGet package scan, `dotnet-inspect`'s `@Integrations` roll-up missed two Azure SDK packages that appear to expose first-class ASP.NET Core Data Protection integration APIs.

### `Azure.Extensions.AspNetCore.DataProtection.Blobs@1.5.3`

Command:

```bash
dnx dotnet-inspect -y -- package Azure.Extensions.AspNetCore.DataProtection.Blobs@1.5.3 --all-libraries -S @Integrations --markdown
```

Observed:

```text
Using TFM: net10.0
# azure.extensions.aspnetcore.dataprotection.blobs 1.5.3
```

No `Integrations` data was reported.

But the package has public ASP.NET Core Data Protection extension APIs:

```text
Microsoft.AspNetCore.DataProtection.AzureStorageBlobDataProtectionBuilderExtensions
└─ PersistKeysToAzureBlobStorage(...): IDataProtectionBuilder
```

`dotnet-inspect` evidence command:

```bash
dnx dotnet-inspect -y -- type '*DataProtection*' --package Azure.Extensions.AspNetCore.DataProtection.Blobs@1.5.3 --all -v:n
```

The nuspec also describes it as "Microsoft Azure Blob storage support as key store" for ASP.NET Core Data Protection, with tags including `aspnetcore dataprotection`, and a dependency on `Microsoft.AspNetCore.DataProtection`.

### `Azure.Extensions.AspNetCore.DataProtection.Keys@1.6.3`

Command:

```bash
dnx dotnet-inspect -y -- package Azure.Extensions.AspNetCore.DataProtection.Keys@1.6.3 --all-libraries -S @Integrations --markdown
```

Observed:

```text
Using TFM: net10.0
# azure.extensions.aspnetcore.dataprotection.keys 1.6.3
```

No `Integrations` data was reported.

But the package has public ASP.NET Core Data Protection extension APIs:

```text
Microsoft.AspNetCore.DataProtection.AzureDataProtectionKeyVaultKeyBuilderExtensions
└─ ProtectKeysWithAzureKeyVault(...): IDataProtectionBuilder
```

`dotnet-inspect` evidence command:

```bash
dnx dotnet-inspect -y -- type '*DataProtection*' --package Azure.Extensions.AspNetCore.DataProtection.Keys@1.6.3 --all -v:n
```

The nuspec also describes it as Azure Key Vault key encryption support, has tags including `aspnetcore dataprotection`, and depends on `Microsoft.AspNetCore.DataProtection`.

## Suggested classification

These probably warrant an ASP.NET Core/Data Protection integration signal. Possible detection patterns:

- public extension methods with first parameter `Microsoft.AspNetCore.DataProtection.IDataProtectionBuilder`
- APIs under `Microsoft.AspNetCore.DataProtection.*` namespaces in non-AspNetCore packages
- packages depending on `Microsoft.AspNetCore.DataProtection` and exposing `IDataProtectionBuilder` extension methods

## Possible noisy/false-positive area from same scan

The same Azure SDK scan also surfaced `Azure.Core@1.57.0` and `System.ClientModel@1.13.0` as DI/Hosting integrations:

```text
Azure.Core: Dependency Injection:1; Hosting:3
System.ClientModel: Dependency Injection:1; Hosting:3
```

Those APIs may be intentional because they expose client registration/credential resolver hosting hooks, but in a broad package scan they read more like foundational/infrastructure integration support than service-specific ecosystem integrations. If this distinction matters, it may be useful to label these differently or document that core infrastructure packages are expected to appear in `@Integrations`.

## dotnet-inspect version

Observed with dotnet-inspect version 0.10.6+30bf7c1.
